netket/graph/grid.py

```diff
@@ -122,28 +186,128 @@ def __repr__(self):
             pbc = self.pbc
         return f"Grid(length={self.length}, pbc={pbc})"
 
-    def periodic_translations(self) -> List[List[int]]:
+    def translations(self, dim: int = None, period: int = 1) -> SymmGroup:
         """
         Returns all permutations of lattice sites that correspond to translations
         along the grid directions with periodic boundary conditions.
 
         The periodic translations are a subset of the permutations returned by
         `self.automorphisms()`.
+
+        Arguments:
+            dim: If set, only translations along `dim` will be returned.
+            period: Period of the translations; should be a divisor of the length in
+                the corresponding lattice dimension.
         """
-        basis = [
-            range(l) if is_per else range(1)
-            for l, is_per in zip(self.length[::-1], self.pbc[::-1])
-        ]
+        dims = tuple(self.length)
+        if dim is None:
+            basis = [
+                range(0, l, period) if is_per else range(1)
+                for l, is_per in zip(dims, self.pbc)
+            ]
+        else:
+            if not self.pbc[dim]:
+                raise ValueError(f"No translation symmetries in non-periodic dim={dim}")
+            basis = [
+                range(0, l, period) if i == dim else range(1)
+                for i, l in enumerate(dims)
+            ]
```

Comment:
Let's support arbitrary dimensions since its cheap
```suggestion
            if np.isscalar(dim):
                dim = [dim]
                
            if not all([self.pbc[d] for d in dim]) :
                raise ValueError(f"No translation symmetries in non-periodic dim={dim}")
                
            basis = [
                range(0, l, period) if i in dim else range(1)
                for i, l in enumerate(dims)
            ]
```